internal/controller/nonadminbackup_controller.go

```diff
@@ -637,7 +638,7 @@ func (r *NonAdminBackupReconciler) createVeleroBackupAndSyncWithNonAdminBackup(c
 			if nabslErr := r.Client.Get(ctx, types.NamespacedName{Name: backupSpec.StorageLocation, Namespace: nab.Namespace}, nonAdminBsl); nabslErr != nil {
 				return false, nabslErr
 			}
-
+			// Should there be a warning that we are overriding the user's storage location?
```

Comment:
this case user created NAB pointing to a NABSL

when creating backup in OADP namespace, we need to change its spec to point to NABSL related BSL

Reply:
if user did not add a name in storageLocation spec field in NAB, we assume is default BSL from OADP namespace (it may not know the name of default BSL there)

if user did add a name in storageLocation spec field in NAB, we assume is a NABSL from its namespace

Reply:
It does not seem possible with current code to create NAB not pointing to NABSL
https://github.com/migtools/oadp-non-admin/blob/f34f8ce1db7c5db01ef022ce5eefc10344f29642/internal/controller/nonadminbackup_controller.go#L632-L643

Reply:
ok got it. empty or nabsl only.